---
title: "ECS markers in oligodendrocyte development"
author: "Diego M. Coelho"
format:
  html:
    toc: true
    toc-depth: 3
    css: "mcblab-style.css"
    self-contained: true
    theme: cosmo  # optional
    sidebar:
      style: "floating"  # or "docked", depending on your preference
      collapse-level: 1
      search: true
editor: visual
---

------------------------------------------------------------------------

![](https://mcblab.github.io/images/logos/lab_icon_long.png){width="200px"} This analysis was performed by [MCB Lab](https://mcblab.github.io/).

```{r libraries}
#| echo: false
#| warning: false

library(Seurat)
library(dplyr)
library(ggplot2)
library(DT)

```

```{r processing}
#| echo: false
#| eval: false
#| warning: false

library(scDblFinder)

# Input can be found:
# https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE255405

GSM_p2_cortex <- ReadMtx(mtx = "GSE255405_RAW/GSM8072107_P2_cortex_matrix.mtx.gz",
                         cells = "GSE255405_RAW/GSM8072107_P2_cortex_barcodes.tsv.gz",
                         features = "GSE255405_RAW/GSM8072107_P2_cortex_features.tsv.gz")

GSM_p2_cortex <- CreateSeuratObject(GSM_p2_cortex, min.cells = 3, min.features = 2)
GSM_p2_cortex@meta.data$time <- "P2"
GSM_p2_cortex@meta.data$area <- "cortex"

GSM_p7_cortex <- ReadMtx(mtx = "GSE255405_RAW/GSM8072108_P7_cortex_matrix.mtx.gz",
                         cells = "GSE255405_RAW/GSM8072108_P7_cortex_barcodes.tsv.gz",
                         features = "GSE255405_RAW/GSM8072108_P7_cortex_features.tsv.gz")

GSM_p7_cortex <- CreateSeuratObject(GSM_p7_cortex, min.cells = 3, min.features = 2)
GSM_p7_cortex@meta.data$time <- "P7"
GSM_p7_cortex@meta.data$area <- "cortex"

GSM_p7_svz <- ReadMtx(mtx = "GSE255405_RAW/GSM8072109_P7_SVZ_matrix.mtx.gz",
                         cells = "GSE255405_RAW/GSM8072109_P7_SVZ_barcodes.tsv.gz",
                       features = "GSE255405_RAW/GSM8072109_P7_SVZ_features.tsv.gz")

GSM_p7_svz <- CreateSeuratObject(GSM_p7_svz, min.cells = 3, min.features = 2)
GSM_p7_svz@meta.data$time <- "P7"
GSM_p7_svz@meta.data$area <- "SVZ"

# Joined version
GSE255405@meta.data$time_area <- paste0(GSE255405@meta.data$time, "_", GSE255405@meta.data$area)

# Joining layers
GSE255405 <- merge(GSM_p2_cortex, list(GSM_p7_cortex, GSM_p7_svz), project = "GSE255405")
GSE255405 <- JoinLayers(GSE255405)

# Removing mitochondrial + low features
GSE255405[["percent.mt"]] <- PercentageFeatureSet(GSE255405, pattern = "^mt-")
GSE255405 <- subset(GSE255405, subset = nFeature_RNA > 200 & percent.mt < 10)

sce <- as.SingleCellExperiment(GSE255405)
sce <- scDblFinder(sce)
GSE255405$scDblFinder.class <- sce$scDblFinder.class

rm(sce)

GSE255405 <- subset(GSE255405, subset = scDblFinder.class == "singlet")

# Normalize
GSE255405 <- NormalizeData(GSE255405)
GSE255405 <- FindVariableFeatures(GSE255405, selection.method = "vst", nfeatures = 2000)

GSE255405 <- ScaleData(GSE255405, features = rownames(GSE255405))



GSE255405 <- CellCycleScoring(GSE255405,
                              s.features = stringr::str_to_title(cc.genes$s.genes),
                              g2m.features = stringr::str_to_title(cc.genes$g2m.genes))

GSE255405 <- RunPCA(GSE255405, features = VariableFeatures(object = GSE255405))

GSE255405 <- FindNeighbors(GSE255405, dims = 1:20)
GSE255405 <- FindClusters(GSE255405, resolution = 0.1)

GSE255405 <- RunUMAP(GSE255405, dims = 1:20)

DimPlot(GSE255405, label = T, pt.size = 1) | DotPlot(GSE255405, features = c("Aldh1l1", "Ascl1", "Aqp4", "Aif1", "Pdgfra", "Enpp6", "Mog", "Tbr1", "Vnn1", "Sp8", "Gad1", "Gad2", "Syt4", "Pecam1", "Foxj1", "Vtn", "Sox2")) + theme(axis.text.x = element_text(angle = 45, hjust = 1))

GSE255405@meta.data$MainGroups <- plyr::mapvalues(GSE255405@meta.data$RNA_snn_res.0.1,
                                                  0:15,
                                                  c("ENs", "Astrocyte", "OPCs/OLs", "TAPs", "INs",
                                                    "INs", "NSCs", "NSCs", "INs", "Microglia",
                                                    "ENs", "Endothelial", "Microglia", "Pericytes", "ENs",
                                                    "ECs"))

Idents(GSE255405) <- "MainGroups"

# Subset OPCs/OLs
GSE255405_OL <- subset(GSE255405, subset = MainGroups == "OPCs/OLs")
GSE255405_OL <- NormalizeData(GSE255405_OL)

GSE255405_OL <- FindVariableFeatures(GSE255405_OL, selection.method = "vst", nfeatures = 2000)
GSE255405_OL <- ScaleData(GSE255405_OL, features = rownames(GSE255405_OL))
GSE255405_OL <- RunPCA(GSE255405_OL, features = VariableFeatures(object = GSE255405_OL))

GSE255405_OL <- FindNeighbors(GSE255405_OL, dims = 1:20)
GSE255405_OL <- FindClusters(GSE255405_OL, resolution = 0.5)

GSE255405_OL <- RunUMAP(GSE255405_OL, dims = 1:10)

DimPlot(GSE255405_OL, label = T, pt.size = 1) | DimPlot(GSE255405_OL, group.by = "area", label = T, pt.size = 1) | DimPlot(GSE255405_OL, group.by = "Phase", label = T, pt.size = 1) | DotPlot(GSE255405_OL, features = c("Pdgfra", "Mbp", "Mog", "Enpp6", "Cspg4", "Npas1", "Dll3", "Ascl1")) + theme(axis.text.x = element_text(angle = 45, hjust = 1))

GSE255405_OL@meta.data$MainGroups <- plyr::mapvalues(GSE255405_OL@meta.data$RNA_snn_res.0.5,
                                                  0:9,
                                                  c("OPCs", "OPCs", "OPCs", "OLs", "OPCs",
                                                    "OPCs", "OLs", "OLs", "OPCs", "OPCs"))

GSE255405_OL@meta.data$MinorGroups <- plyr::mapvalues(GSE255405_OL@meta.data$RNA_snn_res.0.5,
                                                  0:9,
                                                  c("actOPCs", "actOPCs", "hOPCs", "IMyOLs", "hOPCs",
                                                    "ProlOPCs", "IMyOLs", "MyOLs", "ProlOPCs", "ProlOPCs"))

# Readding OPCs / OLs to main GSE255405
GSE255405@meta.data$MinorGroups <- GSE255405@meta.data$MainGroups
GSE255405@meta.data <- GSE255405@meta.data %>% filter(MainGroups != "OPCs/OLs") %>% rbind(., GSE255405_OL@meta.data)
GSE255405@meta.data <- GSE255405@meta.data[GSE255405@graphs$RNA_snn@Dimnames[[1]],]

Idents(GSE255405_OL) <- "MainGroups"

GSE255405_OL.markers.SVZ <- subset(GSE255405_OL, subset = area == "SVZ") %>% FindAllMarkers(., only.pos = T)
GSE255405_OL.markers.cortex <- subset(GSE255405_OL, subset = area != "SVZ") %>% FindAllMarkers(., only.pos = T)

# Save objects
saveRDS(GSE255405, file = "GSE255405.rds")
saveRDS(GSE255405_OL, file = "GSE255405_OL.rds")

```

```{r diff}
#| echo: false
#| eval: false
#| warning: false

GSE255405 <- readRDS("GSE255405_OL.rds")

Idents(GSE255405) <- "MinorGroups"

# Generate markers
GSE255405.SVZ.markers <- subset(GSE255405, subset = area == "SVZ") %>% FindAllMarkers(., only.pos = T)
GSE255405.SVZ.markers <- GSE255405.SVZ.markers %>% filter(p_val_adj < 0.01)
GSE255405.SVZ.markers <- GSE255405.SVZ.markers %>%
  mutate(area = "V/SVZ", age = "P7")

GSE255405.ctx.markers <- subset(GSE255405, subset = area != "SVZ" & time == "P7") %>% FindAllMarkers(., only.pos = T)
GSE255405.ctx.markers <- GSE255405.ctx.markers %>% filter(p_val_adj < 0.01)
GSE255405.ctx.markers <- GSE255405.ctx.markers %>%
  mutate(area = "Cortex", age = "P7")

GSE255405.ctx.p2.markers <- subset(GSE255405, subset = area != "SVZ" & time == "P2") %>% FindAllMarkers(., only.pos = T)
GSE255405.ctx.p2.markers <- GSE255405.ctx.p2.markers %>% filter(p_val_adj < 0.01)
GSE255405.ctx.p2.markers <- GSE255405.ctx.p2.markers %>%
  mutate(area = "Cortex", age = "P2")


saveRDS(GSE255405.SVZ.markers, file = "GSE255405.vsvz.p7.markers.rds")
saveRDS(GSE255405.ctx.markers, file = "GSE255405.ctx.p7.markers.rds")
saveRDS(GSE255405.ctx.p2.markers, file = "GSE255405.ctx.p2.markers.rds")

```

```{r open}
#| echo: false
#| warning: false

GSE255405 <- readRDS("GSE255405.rds")

GSE255405_OL <- readRDS("GSE255405_OL.rds")

markers_vsvz_p7 <- readRDS("GSE255405.vsvz.p7.markers.rds")
markers_ctx_p2 <- readRDS("GSE255405.ctx.p2.markers.rds")
markers_ctx_p7 <- readRDS("GSE255405.ctx.p7.markers.rds")

markers <- rbind(markers_ctx_p2, markers_ctx_p7, markers_vsvz_p7)

glu_genes <- c("Cnr1", "Cnr2", "Cspg4", "Nr3c1", "Nr3c2", "Ptges3", "Ncor1")

```

## Introduction

Early life stress (ELS) profoundly influences brain development by altering oligodendrogenesis and myelination, processes regulated by both glucocorticoids and the endocannabinoid system (ECS). In the referenced study, maternal separation combined with corticosterone exposure accelerated oligodendrocyte lineage progression and myelin marker expression in neonatal mice. These effects were modulated by CB1 receptor activity and 2-AG signaling, highlighting ECS--HPA axis interplay in myelin plasticity. To support these findings, we used the single-cell RNA-seq dataset from Dennis et al. (2024) \[GSE237672\] to better understand the expression dynamics of ECS and glucocorticoid signaling components across oligodendrocyte lineage cells during early postnatal brain development.

Dataset used can be found in [Dennis et al](https://www.sciencedirect.com/science/article/pii/S2213671124000778) publication.

## Questions

1)  Which cell types during early postnatal express glucorticoid receptors and canabinoid receptors as CB1 / CB2?
2)  Does it have a specific OPCs / Oligodendrocytes groups expressing those markers?

## Results

Cell types were annotated following mostly protocol established by Dennis et al publication.

### Which cell types during early postnatal express glucorticoid receptors and canabinoid receptors as CB1 / CB2?

#### UMAP to overview annotated cell groups, time, area and cell phase.

```{r fig1}
#| echo: false
#| fig-width: 16
#| fig-height: 10

p1 <- DimPlot(GSE255405, group.by = "MainGroups", pt.size = 1) + theme(axis.title = element_blank())
p2 <- DimPlot(GSE255405, group.by = "time", pt.size = 1) + theme(axis.title = element_blank())
p3 <- DimPlot(GSE255405, group.by = "area", pt.size = 1) + theme(axis.title = element_blank())
p4 <- DimPlot(GSE255405, group.by = "Phase", pt.size = 1) + theme(axis.title = element_blank())

p <- cowplot::plot_grid(plotlist = list(p1, p2, p3, p4))

# pdf("figures/umap.ctx_vsvz.pdf", width = 7, height = 4)
# p1 + labs(title = "V/SVZ + Cortex (P2/P7)")
# dev.off()

# pdf("figures/umap.ctx_vsvz.age.pdf", width = 6.5, height = 4)
# p2 + labs(title = "V/SVZ + Cortex (P2/P7) - Age")
# dev.off()

# pdf("figures/umap.ctx_vsvz.area.pdf", width = 6.5, height = 4)
# p3 + labs(title = "V/SVZ + Cortex (P2/P7) - Area")
# dev.off()

# pdf("figures/umap.ctx_vsvz.phase.pdf", width = 6.5, height = 4)
# p4 + labs(title = "V/SVZ + Cortex (P2/P7) - Phase")
# dev.off()


p

```

**Figure 1.**  
UMAP visualization of single-cell RNA-seq data from the V/SVZ and cortex at postnatal days P2 and P7.  
Each panel shows cell clustering colored by (A) major cell type groups, (B) developmental time point, (C) brain region (area), and (D) cell cycle phase.  
This overview highlights the main neural and glial populations identified in the dataset.


#### Markers used by author to classify cells

```{r fig2}
#| echo: false
#| warning: false
#| fig-width: 8
#| fig-height: 5

p <- DotPlot(GSE255405, group.by = "MainGroups", features = c("Syt4", "Tbr1", "Aldh1l1", "Aqp4", "Ascl1", "Sp8", "Gad1", "Gad2", "Vnn1", "Sox2", "Vim", "Aif1", "Pecam1", "Vtn", "Foxj1", "Pdgfra", "Enpp6", "Mog")) +
  theme(axis.title = element_blank(), axis.text.x = element_text(angle = 45, hjust = 1))

# pdf("figures/markers.dotplot.ctx_vsvz.pdf", width = 8, height = 3.5)
p + labs(title = "V/SVZ + Cortex (P2/P7) - Markers")
# dev.off()


```

**Figure 2.**  
Dot plot showing expression of canonical marker genes across the main cell populations identified in the V/SVZ and cortex.
Dot size represents the percentage of cells expressing the gene, while color intensity indicates average expression level. 
Markers correspond to neuronal (e.g., *Tbr1*, *Syt4*), astrocytic (*Aldh1l1*, *Aqp4*), progenitor (*Ascl1*, *Sox2*), and oligodendrocyte (*Pdgfra*, *Mog*) lineages.


```{r fig3}
#| echo: false
#| warning: false
#| fig-width: 15
#| fig-height: 5

# pdf("figures/nr3c1_cnrs.dotplot.ctx_vsvz.pdf", width = 8.5, height = 3.5)
p1 <- subset(GSE255405, area != "SVZ" & time == "P2") %>%
DotPlot(features = glu_genes, group.by = "MainGroups") + labs(title = "Cortex (P2)") +
  theme(axis.title = element_blank(), axis.text.x = element_text(angle = 45, hjust = 1))
p2 <- subset(GSE255405, area == "SVZ") %>%
DotPlot(features = glu_genes, group.by = "MainGroups") + labs(title = "V/SVZ (P7)") +
  theme(axis.title = element_blank(), axis.text.x = element_text(angle = 45, hjust = 1))
p3 <- subset(GSE255405, area != "SVZ" & time == "P7") %>%
DotPlot(features = glu_genes, group.by = "MainGroups") + labs(title = "Cortex (P7)") +
  theme(axis.title = element_blank(), axis.text.x = element_text(angle = 45, hjust = 1))
# dev.off()

p1 | p2 | p3

```

**Figure 3.**  
Comparative dot plots illustrating the expression of glucocorticoid and cannabinoid signaling–related genes (*Cnr1*, *Cnr2*, *Nr3c1*, *Nr3c2*, *Ptges3*, *Ncor1*, *Cspg4*) in different brain regions and developmental stages.  
Panels show gene expression in (A) cortex at P2, (B) V/SVZ at P7, and (C) cortex at P7.  
These profiles reveal developmental and spatial differences in the regulatory networks influencing oligodendrocyte lineage cells.


### Does it have a specific OPCs / Oligodendrocytes groups expressing those markers?

```{r fig4}
#| echo: false
#| warning: false
#| fig-width: 12
#| fig-height: 5

p <- DimPlot(GSE255405_OL, group.by = "area", label = T, pt.size = 1) | DimPlot(GSE255405_OL, group.by = "MainGroups", pt.size = 1) | DimPlot(GSE255405_OL, group.by = "MinorGroups", pt.size = 1)

p

```

**Figure 4.**  
UMAP visualization of oligodendrocyte lineage subpopulations (OPCs and OLs) extracted from the combined dataset.  
Cells are colored by their subcluster identity, brain region, and developmental time point.  
This representation highlights the transcriptional continuum from proliferative OPCs to mature myelinating OLs, and the influence of spatial and temporal factors on lineage progression.


```{r fig5}
#| echo: false
#| warning: false
#| fig-width: 15
#| fig-height: 5

# pdf("figures/nr3c1_cnrs.dotplot.ctx_vsvz.pdf", width = 8.5, height = 3.5)
p1 <- subset(GSE255405_OL, area != "SVZ" & time == "P2") %>%
DotPlot(features = glu_genes, group.by = "MinorGroups") + labs(title = "Cortex (P2)") +
  theme(axis.title = element_blank(), axis.text.x = element_text(angle = 45, hjust = 1))
p2 <- subset(GSE255405_OL, area == "SVZ") %>%
DotPlot(features = glu_genes, group.by = "MinorGroups") + labs(title = "V/SVZ (P7)") +
  theme(axis.title = element_blank(), axis.text.x = element_text(angle = 45, hjust = 1))
p3 <- subset(GSE255405_OL, area != "SVZ" & time == "P7") %>%
DotPlot(features = glu_genes, group.by = "MinorGroups") + labs(title = "Cortex (P7)") +
  theme(axis.title = element_blank(), axis.text.x = element_text(angle = 45, hjust = 1))
# dev.off()

p1 | p2 | p3

```

**Figure 5.**  
Dot plot of glucocorticoid and cannabinoid-related genes across oligodendrocytes and OPCs subpopulations.
Dot size corresponds to the proportion of cells expressing each gene, and color intensity reflects average expression level.

# Differential Gene Expression

Table with all markers found within each Time + Area for subpopulations of OPCs + OLs.

| **Column**     | **Description**                                                                                                                                                                                                                  |
| -------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| **Gene**       | The gene symbol or ID of the marker gene identified as differentially expressed. Each entry corresponds to one gene tested across clusters or groups.                                                                            |
| **Cluster**    | The cell cluster (or subpopulation) in which the gene is more highly expressed compared to others. For example, a specific OPC or OL subtype.                                                                                    |
| **Area**       | The **anatomical region** (e.g., cortex, corpus callosum, spinal cord) where this cluster’s cells were sampled. Helps distinguish spatial differences in gene expression.                                                        |
| **Age**        | The **time point** or **developmental stage** (e.g., P7, Adult) at which cells were collected. Allows temporal comparisons of expression changes.                                                                                |
| **avg_log2FC** | The **average log₂ fold-change** in expression between cells in the cluster of interest and all other cells. Positive values indicate upregulation in this cluster; negative values indicate downregulation.                     |
| **pct.1**      | The **percentage of cells in the target cluster** where this gene is detected (nonzero expression). Higher values = more consistently expressed within the cluster.                                                              |
| **pct.2**      | The **percentage of cells in all other clusters** (combined) expressing the same gene. Used to assess specificity of expression.                                                                                                 |
| **p_val_adj**  | The **adjusted p-value** (after multiple testing correction, e.g., Bonferroni or FDR) indicating the statistical significance of the differential expression. Smaller values = stronger evidence that the gene is a true marker. |


```{r tab1}
#| echo: false
#| warning: false

# Create interactive table with download button
markers %>%
  select(
    Gene = gene, Cluster = cluster, Area = area, Age = age,
    avg_log2FC, pct.1, pct.2, p_val_adj) %>%
  mutate(
    avg_log2FC = round(avg_log2FC, 2),   # round to 2 decimals
    p_val_adj = signif(p_val_adj, 3)     # use scientific rounding for p-values
  ) %>%
  datatable(
    rownames = FALSE,
    extensions = 'Buttons',
    options = list(
      dom = 'Bfrtip',
      buttons = c('copy', 'csv', 'excel'),
      scrollX = TRUE,
      pageLength = 10
    ),
    fillContainer = TRUE
  )



```

**Table 1.**  
List of marker genes identified for each OPC/OL subpopulation across different areas and developmental time points.  
The table includes fold-change values, detection percentages, and adjusted p-values for statistical significance.

